include/engine/routing_algorithms/routing_base_mld.hpp

```diff
@@ -680,11 +682,20 @@ double getNetworkDistance(SearchEngineData<Algorithm> &engine_working_data,
         return std::numeric_limits<double>::max();
     }
 
-    std::vector<PathData> unpacked_path;
+    EdgeDistance distance = 0;
 
-    annotatePath(facade, phantom_nodes, unpacked_nodes, unpacked_edges, unpacked_path);
+    if (!unpacked_nodes.empty())
+    {
+        for (auto node_iter = unpacked_nodes.begin(); node_iter != std::prev(unpacked_nodes.end()); node_iter++)
```

Comment:
Why not include the last here?

Reply:
In the edge-based-graph the last edge-based-node is not traveled over (we arrive at the beginning of that street segment).